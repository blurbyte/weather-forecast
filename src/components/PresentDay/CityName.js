import styled from 'styled-components';

const CityName = styled.h1`
  color: ${props => props.theme.primary};
  font-family: 'Cardo', serif;
  line-height: 1;
  font-weight: 700;
  font-size: 4rem;
  letter-spacing: 2px;
  margin: 0;
`;

export default CityName;
