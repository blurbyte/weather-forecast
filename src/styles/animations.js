import { keyframes } from 'styled-components';

export const fadeIn = keyframes`
  from {
    opacity: 0;
    transform: translateY(2rem);
  }
  to {
    opacity: 1;
    transform: translateY(0);
  }
`;
