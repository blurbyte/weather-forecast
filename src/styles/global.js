import { injectGlobal } from 'styled-components';

injectGlobal`
  @import url('https://fonts.googleapis.com/css?family=Cardo:700&subset=latin-ext');

  html {
    font-size: 62.5%;
  }

  body {
    background: #71d7ed;
    color: #222222;
    font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif;
    line-height: 1.4;
    font-size: 1.4rem;
    font-weight: 400;
    overflow-y: scroll;
    user-select: none;
  }
`;
