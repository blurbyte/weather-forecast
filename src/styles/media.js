// Generates media templates based on device breakpoints

import { css } from 'styled-components';

// Devices breakpoints
export const devices = {
  desktop: 1000,
  tablet: 640,
  phone: 420
};

const media = Object.keys(devices).reduce((acc, label) => {
  acc[label] = (strings, ...interpolations) => css`
    @media (max-width: ${devices[label] / 16}em) {
      ${css(strings, ...interpolations)};
    }
  `;
  return acc;
}, {});

export default media;
