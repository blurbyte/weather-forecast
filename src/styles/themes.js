export const dayTheme = {
  primary: '#222222',
  background: '#71d7ed'
};

export const nightTheme = {
  primary: '#ffffff',
  background: '#3033ba'
};
