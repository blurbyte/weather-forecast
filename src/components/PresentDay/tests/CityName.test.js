import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { dayTheme } from '../../../styles/themes';
import CityName from '../CityName';

test('renders correctly', () => {
  const component = renderer.create(<CityName theme={dayTheme} />);
  expect(component.toJSON()).toMatchSnapshot();
});
