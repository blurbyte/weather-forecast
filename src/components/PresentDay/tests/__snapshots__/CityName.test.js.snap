// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  color: #222222;
  font-family: 'Cardo',serif;
  line-height: 1;
  font-weight: 700;
  font-size: 4rem;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  margin: 0;
}

<h1
  className="c0"
/>
`;
